File: python/8kyu/bonus_time.py
import codewars_test as test


def bonus_time(salary: int, bonus: bool) -> str:
    return f"${salary*10}" if bonus else f"${salary}"


@test.describe("Fixed Tests")
def basic_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(bonus_time(10000, True), '$100000')
        test.assert_equals(bonus_time(25000, True), '$250000')
        test.assert_equals(bonus_time(10000, False), '$10000')
        test.assert_equals(bonus_time(60000, False), '$60000')
        test.assert_equals(bonus_time(2, True), '$20')
        test.assert_equals(bonus_time(78, False), '$78')
        test.assert_equals(bonus_time(67890, True), '$678900')

File: typescript/8kyu/bonus_time/bonus_time.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Kata = void 0;
class Kata {
    static bonusTime(salary, bonus) {
        return `£${bonus ? salary * 10 : salary}`;
    }
}
exports.Kata = Kata;
// import the type of assertion library you wish to use (Chai recommended)
const chai_1 = require("chai");
describe("Fixed Tests", function () {
    it("Basic tests", function () {
        chai_1.assert.equal(Kata.bonusTime(10000, true), '£100000');
        chai_1.assert.equal(Kata.bonusTime(10000, false), '£10000');
        chai_1.assert.equal(Kata.bonusTime(25000, true), '£250000');
        chai_1.assert.equal(Kata.bonusTime(60000, false), '£60000');
        chai_1.assert.equal(Kata.bonusTime(2, true), '£20');
        chai_1.assert.equal(Kata.bonusTime(78, false), '£78');
        chai_1.assert.equal(Kata.bonusTime(67890, true), '£678900');
    });
});
